src/Http/Client.php

```diff
@@ -192,6 +193,9 @@ public function cookies()
      */
     public function addCookie(CookieInterface $cookie)
     {
+        if (!$cookie->getDomain() || !$cookie->getPath()) {
+            throw new InvalidArgumentException('Cookie must have a domain and a path set.');
+        }
```

Comment:
This might caused otherwise valid cookies to be rejected. For example

```
Set-Cookie: foo=bar
```

Is a perfectly legal cookie header, which would now be 'invalid' breaking client requests to servers that make cookies like this. Could we instead fix the needle check?

Reply:
Ignore me. When cookie values are parsed we apply the request path/domain if the cookie does not specify them.

Reply:
This is a HTTP Client cookies collection. No response cookies involved.

Reply:
Good point. I missed that.